common/shareddb.cpp

```diff
@@ -65,40 +66,56 @@ SharedDatabase::~SharedDatabase() = default;
 
 bool SharedDatabase::SetHideMe(uint32 account_id, uint8 hideme)
 {
-	const std::string query = StringFormat("UPDATE account SET hideme = %i WHERE id = %i", hideme, account_id);
-	const auto results = QueryDatabase(query);
-	if (!results.Success()) {
-		return false;
+	auto a = AccountRepository::FindOne(*this, account_id);
+	if (a.id > 0) {
+		a.hideme = hideme ? 1 : 0;
+		AccountRepository::UpdateOne(*this, a);
 	}
 
-	return true;
+	return a.id > 0;
 }
 
 uint8 SharedDatabase::GetGMSpeed(uint32 account_id)
 {
-	const std::string query = StringFormat("SELECT gmspeed FROM account WHERE id = '%i'", account_id);
-	auto results = QueryDatabase(query);
-	if (!results.Success()) {
-		return 0;
+	auto a = AccountRepository::FindOne(*this, account_id);
+	if (a.id > 0) {
+		return a.gmspeed;
 	}
 
-    if (results.RowCount() != 1)
-        return 0;
+	return 0;
+}
 
-    auto& row = results.begin();
+bool SharedDatabase::SetGMSpeed(uint32 account_id, uint8 gmspeed)
+{
+	auto a = AccountRepository::FindOne(*this, account_id);
+	if (a.id > 0) {
+		a.gmspeed = gmspeed ? 1 : 0;
+		AccountRepository::UpdateOne(*this, a);
+	}
 
-	return atoi(row[0]);
+	return a.id > 0;
 }
 
-bool SharedDatabase::SetGMSpeed(uint32 account_id, uint8 gmspeed)
+bool SharedDatabase::SetGMInvul(uint32 account_id, bool gminvul)
 {
-	const std::string query = StringFormat("UPDATE account SET gmspeed = %i WHERE id = %i", gmspeed, account_id);
-	const auto results = QueryDatabase(query);
-	if (!results.Success()) {
-		return false;
+	auto a = AccountRepository::FindOne(*this, account_id);
+	if (a.id > 0) {
+		a.invulnerable = gminvul ? 1 : 0;
+		AccountRepository::UpdateOne(*this, a);
 	}
 
-	return true;
+	return a.id > 0;
+}
+
+bool SharedDatabase::SetGMFlymode(uint32 account_id, uint8 flymode)
+{
+	auto a = AccountRepository::FindOne(*this, account_id);
+	if (a.id > 0) {
+		a.flymode = flymode ? 1 : 0;
+		AccountRepository::UpdateOne(*this, a);
+	}
+
+	return a.id > 0;
```

Comment:
Does this allow flymodes other than just Fly Mode 1 (Flying)? There's 4 other types as well.

Reply:
Fixed this one